provider/datagen/src/lib.rs

```diff
@@ -185,6 +205,39 @@ impl DatagenProvider {
         TEST_PROVIDER.clone()
     }
 
+    pub(crate) fn filter_data_locales(
+        &self,
+        supported: Vec<icu_provider::DataLocale>,
+    ) -> Vec<icu_provider::DataLocale> {
+        match &self.source.options.locales {
+            options::LocaleInclude::All => supported,
+            options::LocaleInclude::Explicit(set) => supported
+                .into_iter()
+                .filter(|l| {
+                    if let Some(fallbacker) = &self.source.fallbacker {
+                        // Include any UND-*
+                        if l.get_langid().language == Language::UND {
+                            return true;
+                        }
+                        let mut chain = fallbacker
+                            .for_config(Default::default())
```

Comment:
Nit: The function should take the DataKey and you should use it when configuring the fallbacker